Std/Data/Fin/Basic.lean

```diff
@@ -49,6 +49,12 @@ def natAdd (n) (i : Fin m) : Fin (n + m) := ⟨n + i, Nat.add_lt_add_left i.2 _
 /-- `min n m` as an element of `Fin (m + 1)` -/
 def clamp (n m : Nat) : Fin (m + 1) := ⟨min n m, Nat.lt_succ_of_le (Nat.min_le_right ..)⟩
 
+/-- `enum n` is the array of all elements of `Fin n` in order -/
+def enum (n) : Array (Fin n) := Array.ofFn id
+
+/-- `list n` is the list of all elements of `Fin n` in order -/
+def list (n) : List (Fin n) := (enum n).data
```

Comment:
not going to block the PR for this, but I think this can be done in one pass without an additional array allocation.

Reply:
Yes, but that makes it difficult to prove anything. It might make sense to have a csimp lemma though.

Reply:
It's a bit weird for this list function to be defined in terms of array functions, we should have enough to do the proof directly over lists

Reply:
I agree. The issue is proving it _here_, this would be much easier with `import Std`...

I tried a few different approaches and this is the one that had was the least unpleasant.